pontoon/base/views.py

```diff
@@ -427,6 +433,106 @@ def entities(request):
     }, safe=False)
 
 
+@login_required(redirect_field_name='', login_url='/403')
+@require_POST
+@require_AJAX
+@transaction.atomic
+def batch_edit_translations(request):
```

Comment:
I'm not sure if it wouldn't be such bad idea to consider moving some parts of this view into celery tasks because there's a chance that parts like calculation of stats will take a while.